src/analysis/TopologyUpdate.jl

```diff
@@ -6,22 +6,32 @@ import REPL: ends_with_semicolon
 function updated_topology(old_topology::NotebookTopology, notebook::Notebook, cells)
 	
 	updated_codes = Dict{Cell,ExprAnalysisCache}()
+	updated_nodes = Dict{Cell,ReactiveNode}()
 	for cell in cells
 		if !(
 			haskey(old_topology.codes, cell) && 
 			old_topology.codes[cell].code === cell.code
 		)
-			updated_codes[cell] = ExprAnalysisCache(notebook, cell)
+			parsedcode = parse_custom(notebook, cell)
+			new_node = parsedcode |>
+				ExpressionExplorer.try_compute_symbolreferences |>
+				ReactiveNode
+			using_imports = ExpressionExplorer.compute_usings_imports(parsedcode)
+			new_code = ExprAnalysisCache(
+				code=cell.code,
+				parsedcode=parsedcode,
+				module_usings_imports=using_imports,
+				function_wrapped=isempty(filter(funcname -> !startswith(string(funcname), "anon"), new_node.funcdefs_without_signatures)) &&
+					:eval ∉ new_node.references && :include ∉ new_node.references &&
+					isempty(using_imports.usings) && isempty(using_imports.imports) &&
```

Comment:
the eval, include, using, import checks are already done by `can_be_function_wrapped`, right? I think the `can_be_function_wrapped` is super fast, so the optimization is probably not necessary.